akka-actor-typed-tests/src/test/scala/docs/akka/typed/coexistence/UntypedWatchingTypedSpec.scala

```diff
@@ -54,7 +54,7 @@ object UntypedWatchingTypedSpec {
   object Typed {
     sealed trait Command
     final case class Ping(replyTo: ActorRef[Pong.type]) extends Command
-    case object Pong
+    final case object Pong
```

Comment:
no need for `final` on objects, good to use it on `case class` though

only obscure reason I would see for using final on an object if you want to prevent extension of `Pong$` from Java, but already using something with $ from Java is a bad smell

Reply:
In another snippet, it's final case object, The main reason here for me is for alignment.
and the final case object is a good practice too when hard to work with Java.
The other snippet IIRC is in the Akka tyepd docs too.

Reply:
please remove the `final` in the other sample then

the Scala example code should use Scala style, and there is a convention to use `final` on `case class` because extending such cause trouble and pattern match is working better

`object` is effectively final anyway